### PR TITLE
feat: agent - eBPF Persist Java symbol table

### PR DESCRIPTION
**Pull request by yinjiping**
**Implement persistent Java symbol retrieval, as follows:**

- Use Unix socket for obtaining Java symbols uniformly, whether it's a Java application deployed in a container or running directly on a node machine.
- Adapt Java symbol data retrieval for agent restart scenarios; Java replay data is reacquired upon agent restart.
- Configure the Java agent.so socket in non-blocking mode to prevent blocking Java threads.
- Ensure timely update of Java symbol data to files to avoid prolonged residence in cache.
- Manage symbol updates for each thread of a Java process.
- Use epoll instead of select() to avoid exceeding the maximum file handle limit (1024).
- [Add check for HotSpot/OpenJ9 JVM dependency file](https://github.com/deepflowio/deepflow/pull/7325/commits/cb8dd0ce8fce0b1c8246a2aff6ce109b59ea7dce)
- [Because it's persistent, multiple agents cannot run simultaneously](https://github.com/deepflowio/deepflow/pull/7325/commits/a4d5af83ff2762b8aad111ec290f839c9ff2290e)
- [To detect the disconnection of the target Java program and output logs](https://github.com/deepflowio/deepflow/pull/7325/commits/402d2d41dab8cd1720671c13b0e22a290887dde3)
- [Add handling for the 'JVMTI_EVENT_COMPILED_METHOD_UNLOAD' event](https://github.com/deepflowio/deepflow/pull/7325/commits/ae0816a02f6286ead255d5acdbce34a263dc385a)
- [Unified Log Output](https://github.com/deepflowio/deepflow/pull/7325/commits/89fa4e630ccd21a7920fcdc8ebed7efb093f9275)
- [Implementing thread pool management mechanism](https://github.com/deepflowio/deepflow/pull/7325/commits/3fa0c6959f5ba5023f22d0baff7dc541c3c06a5a)
- [After attaching, clean up the residual files in the target JAVA namespace](https://github.com/deepflowio/deepflow/pull/7325/commits/0a3e20db38d14b4b844a8f472682eaf611e22920) 
- [Adjustments to Profile JAVA-related configuration items](https://github.com/deepflowio/deepflow/pull/7325/commits/c0f1e10af2784967eaf78e392e601d4ca37c5396)
- [Modify Java-symbol file generation call interface and adjust the compilation](https://github.com/deepflowio/deepflow/pull/7325/commits/0faa81daa4990288c7766d6c6e3925282abfddc0) 

### This PR is for:

- Agent

#### Affected branches
- main
- v6.5

### CODE REVIEW

File: server/agent_config/example.yaml [hunk 1]
```diff
@@ -1396,23 +1396,13 @@ vtap_group_id: g-xxxxxx
...
-    ## Maximum size limit for Java symbol file.
-    ## Default: 10. Range: [2, 100]
-    ## Note:
-    ##   Which means it falls within the interval of 2Mi to 100Mi. If the configuration value is outside
-    ##   this range, the default value of 10(10Mi), will be used.
-    ##   All Java symbol files are stored in the '/tmp' directory mounted by the deepflow-agent. To prevent
-    ##   excessive occupation of host node space due to large Java symbol files, a maximum size limit is set
-    ##   for each generated Java symbol file.
-    #java-symbol-file-max-space-limit: 10
+    #java-symbol-file-refresh-defer-interval: 60s
```
Comment (yinjiping) [inline server/agent_config/example.yaml:1405]: 这里去掉了 “java-symbol-file-max-space-limit” 这个配置选项，java-symbol-file-refresh-defer-interval 默认值修改了。 @yuanchaoa 帮忙review看看有什么遗漏

Comment (sharang) [reply server/agent_config/example.yaml:1405]: @yinjiping 最小值是不是可以改为 1，即 1s

Comment (sharang) [reply server/agent_config/example.yaml:1405]: agent 刚启动，或者 java 进程刚启动时，会有滞后处理吗？注释中描述一下这个滞后的行为

File: agent/src/ebpf/user/profile/perf_profiler.h [hunk 2]
```diff
@@ -22,6 +22,8 @@
 #include "../../kernel/include/perf_profiler.h"
 
 #define PROFILER_CTX_NUM 2
+// For storing information about continuously running profiling processes.
+#define PROFILER_RUNNING_PID_PATH "/tmp/.deepflow-profiler-pid"
```
Comment (sharang) [inline agent/src/ebpf/user/profile/perf_profiler.h:26]: deepflow-profiler 是个新进程吗？它应该叫 deepflow-jvm-symbol-collector 之类？
它的主要功能是什么

File: server/agent_config/example.yaml (hunk 1 above)
Comment (yinjiping) [reply server/agent_config/example.yaml:1405]: > @yinjiping 最小值是不是可以改为 1，即 1s

持续剖析还有一层符号缓存，当Java符号文件生成完毕后，解析缓存会就去掉所有的之前记录，重新建立缓存并把文件中所有符号逐个加入到缓存中，这个过程比较消耗CPU资源，设置时间太小的话刷新频率高。

Comment (yinjiping) [reply server/agent_config/example.yaml:1405]: > agent 刚启动，或者 java 进程刚启动时，会有滞后处理吗？注释中描述一下这个滞后的行为

会有滞后处理，刚启动编译的符号还没有预热，method unload事件比较多，需要给一个预热时间。我加些描述

Comment (sharang) [reply server/agent_config/example.yaml:1410]: deepflow-agent 自己启动时的滞后行为也描述下？ @yinjiping

**Review by rvql — APPROVED** [review]

**Review by sharang — APPROVED** [review]

**Pull request merged**